File: adventofcode/2020/09/9.cpp
#include <bits/stdc++.h>
using namespace std;

int n, o;
vector<long long> a, sum;

long long find_pair() {
    for (int i = o; i<n; i++) {
        bool found = false;
        for (int j = i-o; j < i-1; j++)
            for (int k = j+1; k < i; k++) 
                if (a[j] + a[k] == a[i])
                    found = true;
        if (!found) return a[i];
    }
    return -1;
}

long long find_cont(int acc) {
    for(int i = 0; i < n-1; i++) {
        for(int j = i+2; j < n+1; j++){
            if (sum[j] - sum[i] == acc){
                long long m = a[i], M = a[i];
                for (int k = i; k<j; k++) {
                    m = min(a[k], m);
                    M = max(a[k], M);
                }
                return m + M;
            }
        }
    }
    return -1;
}

int main() {
    cin >> n >> o;
    a = vector<long long>(n);
    sum.push_back(0);
    for (int i = 0; i < n; i++){
        cin >> a[i];
        sum.push_back(sum.back()+a[i]);
    }
    int sol = find_pair();
    cout << "Problem1: " << sol << endl << "Problem2: " << find_cont(sol) << endl;
}

File: adventofcode/2020/09/9.py
# slower but easier to code
from itertools import product
def find_cont():
    for i in range(2, n+1):
        for j in range(0, n-i):
            if sum(a[j:j+i]) == sol1:
                return max(a[j:j+i]) + min(a[j:j+i])

N = int(input())
O = int(input())
a = [int(input()) for _ in range(0, N)]
for n in range(O, N):
    found = False
    for i, j in product(a[n-O:n], a[n-O:n]):
        if i != j and i + j == a[n]:
            found = True
            break
    if not found:
        sol1 = a[n]
        break
sol2 = find_cont()
print("Problem1: {}\nProblem2: {}".format(sol1, sol2))